nixos/modules/programs/vscode.nix

```
@@ -0,0 +1,68 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.vscode;
+in
+{
+  options.programs.vscode = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = "Whether to enable Vscode.";
+    };
+
+    defaultEditor = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        When enabled, installs Vscode and configures Vscode to be the default editor
```

```suggestion
        When enabled, configures Vscode to be the default editor
```
I hope the enable option installs the editor